test/e2e/resourceinterpreter_test.go

```diff
@@ -121,4 +123,74 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
 			framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
 		})
 	})
+
+	ginkgo.Context("Workload resource ReviseReplica hook testing", func() {
+		policyNamespace := testNamespace
+		policyName := workloadNamePrefix + rand.String(RandomStrLength)
+		workloadNamespace := testNamespace
+		workloadName := policyName
+		workload := testhelper.NewWorkload(workloadNamespace, workloadName)
+		workload.Spec.Replicas = pointer.Int32Ptr(4)
+
+		policy := testhelper.NewPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: workload.APIVersion,
+				Kind:       workload.Kind,
+				Name:       workload.Name,
+			},
+		}, policyv1alpha1.Placement{
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: framework.ClusterNames(),
+			},
+			ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+				ReplicaDivisionPreference: policyv1alpha1.ReplicaDivisionPreferenceWeighted,
+				ReplicaSchedulingType:     policyv1alpha1.ReplicaSchedulingTypeDivided,
+				WeightPreference: &policyv1alpha1.ClusterPreferences{
+					StaticWeightList: []policyv1alpha1.StaticClusterWeight{
+						{
+							TargetCluster: policyv1alpha1.ClusterAffinity{
+								ClusterNames: []string{"member1"},
+							},
+							Weight: 2,
+						},
+						{
+							TargetCluster: policyv1alpha1.ClusterAffinity{
+								ClusterNames: []string{"member2"},
+							},
+							Weight: 1,
+						},
+						{
+							TargetCluster: policyv1alpha1.ClusterAffinity{
+								ClusterNames: []string{"member3"},
+							},
+							Weight: 1,
+						},
```

Comment:
How about calculating the replicas like this:

https://github.com/karmada-io/karmada/blob/14487548dd9c4970dad003da7d253c25811f2515/test/e2e/scheduling_test.go#L535-L546

In this way, we can make the logic is independent of the specific cluster name.

I also plan to gradually change other E2E parts that involve specific cluster names to those that are not related to cluster names.

Reply:
Great advice, I was wondering how to avoid using specific cluster names myself, updated, PTAL

Reply:
I don't have a specific idea yet. It may be determined based on the cluster type or traverse the existing cluster.
If you have a good idea, we can discuss it :)

Reply:
For my case now, I think the way in the scheduling test is just fine.